default/python/AI/MoveUtilsAI.py

```diff
@@ -86,11 +86,26 @@ def can_travel_to_system(fleet_id, from_system_target, to_system_target, ensure_
             print "target in Ring 2, has enough aggression"
         elif target_sys_id in ColonisationAI.annexable_ring3 and foAI.foAIstate.character.may_travel_beyond_supply(3):
             print "target in Ring 2, has enough aggression"
-    if (not unsupplied_stops or not ensure_return or
-        target_sys_id in fleet_supplyable_system_ids and len(unsupplied_stops) <= fuel
-        or target_sys_id in ColonisationAI.annexable_ring1 and len(unsupplied_stops) < fuel
-        or target_sys_id in ColonisationAI.annexable_ring2 and foAI.foAIstate.character.may_travel_beyond_supply(2) and len(unsupplied_stops) < fuel - 1
-        or target_sys_id in ColonisationAI.annexable_ring3 and foAI.foAIstate.character.may_travel_beyond_supply(3) and len(unsupplied_stops) < fuel - 2):
+
+    def has_enough_fuel(distance_from_supply):
+        # Let f be the ships fuel
+        # Let n be the number of unsupplied stops
+        # Let d be the distance from target system to supply
+        #
+        # After reaching our target, we have f_new = f_current - n
+        # We then need another d fuel to resupply,
+        # i.e. f_new >= d or f_current >= (n + d)
+        return fuel > (len(unsupplied_stops) + distance_from_supply)
+
+    def movement_allowed(x):
```

Comment:
It's a pretty minor thing, but it sort of grates on me to see a variable name x for something that's always expected to be an integer-- howabout 'n' or 'i' or even 'jumps'?  Or, well actually I am going to propose a more complicated calc so I think we would need two such vars, perhaps m and n, though really more descriptive names would be better.